packages/webapp/src/test-utils/storybook/unit.jsx

```diff
@@ -0,0 +1,174 @@
+    this.select = this.canvas.getByTestId(`${testId}-select`);
+    this.testId = testId;
+    this.unitType = unitType;
+  }
+
+  async clearInput() {
+    await userEvent.clear(this.visibleInput);
+  }
+
+  async clearInputAndBlur() {
+    await this.clearInput();
+    await userEvent.click(document.body);
+  }
+
+  async clearError() {
+    const clearButton = await this.canvas.findByTestId(`${this.testId}-errorclearbutton`);
+    await userEvent.click(clearButton);
+  }
+
+  async inputValue(value) {
+    await this.clearInput();
+    await userEvent.type(this.visibleInput, value);
+  }
+
+  async inputValueAndBlur(value) {
+    await this.inputValue(value);
+    await userEvent.click(document.body);
+  }
+
+  async selectUnit(unit) {
+    await selectEvent.openMenu(within(this.select).getByRole('combobox'));
+    await selectEvent.select(within(this.select).getByRole('combobox'), unit);
+  }
+
+  async testSelectedUnit(selectedUnit) {
+    const unit = await within(this.select).findByText(selectedUnit);
+    expect(unit).toBeInTheDocument();
+  }
+
+  async testVisibleValue(value) {
+    await waitFor(() => {
+      expect(this.visibleInput).toHaveValue(value);
+    });
+  }
+
+  /**
+   * Test hidden value.
+   * @param {number} value - Hidden value to test or visible value to convert.
+   * @param {string} [selectedUnit] - Selected unit. Should not be passed with hidden value.
+   * @param {string} [databaseUnit] - Database unit. Should not be passed with hidden value.
+   * @return void
+   */
+  async testHiddenValue(value, selectedUnit, databaseUnit) {
+    let hiddenValue = value;
+    if (value && selectedUnit && databaseUnit) {
+      hiddenValue = convert(value).from(selectedUnit).to(databaseUnit);
+    }
+    await waitFor(() => expect(this.hiddenInput).toHaveValue(hiddenValue));
+  }
+
+  async testNoInput() {
+    await waitFor(() => {
+      console.log(this.visibleInput, this.hiddenInput);
+      expect(this.visibleInput).toHaveValue(null);
+      expect(this.hiddenInput).toHaveValue(null);
+    });
+  }
+
+  async testDisabledStatus() {
+    await waitFor(() => {
+      expect(this.visibleInput).toBeDisabled();
+      expect(within(this.select).getByRole('combobox')).toBeDisabled();
+    });
+  }
+
+  async testRequiredError() {
+    const errorContainer = await this.canvas.findByTestId(`${this.testId}-errormessage`);
+    expect(errorContainer).toHaveTextContent('Required');
+  }
+
+  async testMaxValueError() {
+    const errorContainer = await this.canvas.findByTestId(`${this.testId}-errormessage`);
+    expect(errorContainer).toHaveTextContent(/Please enter a value between 0-[0-9]./);
+  }
+
+  async testNoError() {
+    await waitFor(() => {
+      const clearButton = this.canvas.queryByTestId(`${this.testId}-errorclearbutton`);
+      const requiredErrorElement = this.canvas.queryByText('Required');
+      const maxValueErrorElement = this.canvas.queryByTestId(`${this.testId}-errorclearbutton`);
```

Comment:
@SayakaOno 

Should these be await this.canvas.findByTestId(`${this.testId}-errormessage`) like above? 

It seems like clearButton = maxValueErrorElement here

Reply:
There are three ways to query elements in testing-library:
- getBy... (find an existing element)
- findBy... (used with `await` to wait for an element to show up)
- queryBy... ("find" an element that does not exist)

I used `waitFor` and `queryByTestId` here to wait for errors to be gone.
Here is the reference: https://testing-library.com/docs/guide-disappearance/

Thank you finding the mistake...!